lib/revision-selector.jsx

```diff
@@ -1,56 +1,89 @@
-import React from 'react'
+import React, { PropTypes } from 'react'
 import moment from 'moment'
+import { property } from 'lodash';
 
-export default React.createClass( {
+const lastUpdateOf = property( 'data.modificationDate' );
+const revisionSorter = ( a, b ) => lastUpdateOf( a ) - lastUpdateOf( b );
 
-	getDefaultProps: function() {
+const sortedRevisions = revisions => (
+	revisions
+		.slice()
+		.sort( revisionSorter )
+);
+
+export const RevisionSelector = React.createClass( {
+	mixins: [
+		require( 'react-onclickoutside' )
+	],
+
+	getInitialState() {
 		return {
-			revisions: []
+			revisions: sortedRevisions( this.props.revisions ),
+			selection: Infinity,
 		}
 	},
 
-	componentDidMount: function() {
-		// Note: this is intentionally not in state so that the UI doesn't update after resetSelection()
-		this.selection = -1;
+	componentWillReceiveProps( { revisions: nextRevisions } ) {
+		const { revisions: prevRevisions } = this.props;
+
+		if ( nextRevisions === prevRevisions ) {
+			return;
+		}
+
+		this.setState( {
+			revisions: sortedRevisions( nextRevisions ),
+		} );
 	},
 
-	mixins: [
-		require( 'react-onclickoutside' )
-	],
+	componentDidUpdate( { revisions: prevRevisions } ) {
+		const { revisions: nextRevisions } = this.props;
+
+		if ( prevRevisions !== nextRevisions ) {
+			// I'm not sure why exactly, but
+			// this control wasn't refreshing
+			// after loading in the revisions
+			// the first time.
```

Comment:
@nfcampos I think what's happening is simply that it's not rendering. you can log the state changes and the render; you can actually open up the HTML view in the dev tools and watch the `<input value={ value } />` change _in the rendered HTML_ and for some reason it doesn't update visually.

As much as I am hesitant to think of these explanations, I suspect some bug in the `input` component or with React rendering it. I noticed some related issues in IE11 with `input` and plain-old HTML

Thanks as always for your feedback!